src/components/postgresConnect.ts

```diff
@@ -1,45 +1,210 @@
+    try {
+      const result = await this.query(
+        `INSERT into action_queue (${Object.keys(
+          action
+        )}, time_queued) VALUES (${this.getValuesPlaceholders(action)}, CURRENT_TIMESTAMP)`,
+        Object.values(action)
+      )
+    } catch (err) {
+      console.log(err.stack)
+    }
+  }
+
+  public executeActionQueued = async (payload: ActionQueueExecutePayload) => {
+    try {
+      await this.query(
+        'UPDATE action_queue SET status = $1, error_log = $2, execution_time = CURRENT_TIMESTAMP WHERE id = $3',
+        Object.values(payload)
+      )
+    } catch (err) {
+      return err.stack
+    }
+  }
+
+  public getActionsQueued = async (
+    payload: ActionQueueGetPayload = { status: 'Scheduled' }
+  ): Promise<ActionQueue[]> => {
+    try {
+      const result = await this.query(
+        'SELECT id, action_code, parameters, execution_time FROM action_queue WHERE status = $1 ORDER BY execution_time',
+        Object.values(payload)
+      )
+      return result.rows as ActionQueue[]
+    } catch (err) {
+      return err.stack
+    }
+  }
 
-pgClient.on('notification', (msg: DatabaseRecord) => {
-  switch (msg.channel) {
-    case 'trigger_notifications':
-      processTrigger(pgClient, JSON.parse(msg.payload))
-      break
-    case 'action_notifications':
-      executeAction(pgClient, JSON.parse(msg.payload), actionLibrary)
-      break
+  public addFile = async (payload: FilePayload): Promise<Pick<File, 'id'>> => {
+    try {
+      const result = await this.query(
+        `INSERT INTO file (${Object.keys(payload)}) VALUES () RETURNING id`,
+        Object.values(payload)
+      )
+      return result.rows[0] as Pick<File, 'id'>
+    } catch (err) {
+      return err.stack
+    }
   }
-  // console.log(msg.payload);
-})
 
-pgClient.query('LISTEN trigger_notifications')
-pgClient.query('LISTEN action_notifications')
+  public getFile = async (payload: FileGetPayload): Promise<File> => {
+    try {
+      const result = await this.query(
+        'SELECT path, original_filename FROM file WHERE id = $1',
+        Object.values(payload)
+      )
+      return result.rows[0] as File
+    } catch (err) {
+      return err.stack
+    }
+  }
+
+  public addActionPlugin = async (plugin: ActionPluginPayload) => {
+    try {
+      await this.query(
+        `INSERT INTO action_plugin (${Object.keys(plugin)}) VALUES (${this.getValuesPlaceholders(
+          plugin
+        )})`,
+        Object.values(plugin)
+      )
+    } catch (err) {
+      console.log(err.stack)
+    }
+  }
+
+  public deleteActionPlugin = async (payload: ActionPluginDeletePayload) => {
+    try {
+      await this.query('DELETE FROM action_plugin WHERE code = $1', Object.values(payload))
+    } catch (err) {
+      console.log(err.stack)
+    }
+  }
+
+  public getActionPlugins = async (): Promise<ActionPlugin[]> => {
+    try {
+      const result = await this.query('SELECT * FROM action_plugin', [])
```

Comment:
Is the empty array necessary? (I thought it you just pass a single query parameter if there are no parameter substitutions).

Reply:
Ah, yea. I can fix that now. : )